src/views/tutorial-view/utils/use-progress-toast.ts

```diff
@@ -0,0 +1,107 @@
+import { useEffect } from 'react'
+import { useCollectionProgress, useTutorialProgress } from 'hooks/progress'
+import usePrevious from 'hooks/use-previous'
+import { Collection, Tutorial } from 'lib/learn-client/types'
+import makeProgressToast from './make-progress-toast'
+
+/**
+ * Given a tutorialId, collectionId, and count of tutorials in the collection,
+ *
+ * Show progress toast when the specified tutorial changes progress,
+ * from some non-"complete" status to "complete".
+ *
+ * This toast will give information related to collection progress,
+ * which is why we need the array of tutorial IDs in the collection.
+ */
+export function useProgressToast({
```

Comment:
This is similar in many ways to the previous implementation in `hashicorp/learn`. I've leaned into comments here to explain the early returns from the effect and so on, but happy to dive in further if needed 🤿